### PR TITLE
raftstore: clean up metadata on start

### PR DESCRIPTION
**Pull request by BusyJay**
If a region removes a peer on a store, it usually puts a tombstone mark then deletes all meta and data. In some condition, TiKV may exit before clearing meta and data (like killed, core or system shutdown). When it's restarted, data will be cleaned up thanks to our current mechanism, but the meta will remain on disk. If the same region schedules a new peer on it again, the old meta will be picked up, hence we get a corrupted region peer.

This pr tries to resolve it by clearing meta too on start.

### CODE REVIEW

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -287,6 +293,18 @@ impl<T, C> Store<T, C> {
         Ok(())
     }
 
+    fn clean_up_meta(&mut self, wb: &mut WriteBatch, region: &metapb::Region) {
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: `clean_up_meta` here confuses me, why not naming clear directly? and this function seems not only clearing the meta but also writing a tombstone key.

Comment (BusyJay) [inline src/raftstore/store/store.rs]: Why does `clean_up_meta` confuse you? What do you think `clean_up_meta` do? Actually the tombstone key exists already, but `peer_storage::clear_meta` will delete it, hence write it back again.

Comment (siddontang) [inline src/raftstore/store/store.rs]: I think it is not easy to distinguish `clean_up` and `clear`.

Comment (BusyJay) [inline src/raftstore/store/store.rs]: We have a function named `clean_up` already. Maybe rename two method to `clear_meta` and `clear_data` respectively?

Comment (siddontang) [inline src/raftstore/store/store.rs]: That may make more sense. 👍

**Comment by zhangjinpeng1987** [comment]
@javaforfun PTAL

**Pull request merged**